src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.ML
     /// </summary>
     public abstract class TrainCatalogBase
     {
-        protected internal readonly IHost Host;
+        protected internal IHostEnvironment Host { get; set; }
```

Comment:
>protected internal [](start = 8, length = 18)

`private protected` FYI

Reply:
removed the Host property

---
In reply to: [256216915](https://github.com/dotnet/machinelearning/pull/2524#discussion_r256216915) [](ancestors = 256216915)